src/D_functions_and_procedures.sql

```diff
@@ -23,6 +23,44 @@ LANGUAGE 'plpgsql';
 --D2 TEST
 SELECT * FROM simple_search('apple',1);
 
+-- Structured string search
+CREATE OR REPLACE FUNCTION structured_string_search (
+  p_title VARCHAR(100), 
+  p_plot VARCHAR(100), 
+  p_character VARCHAR(100), 
+  p_person VARCHAR(100), 
+  p_user_id INTEGER
+)
+RETURNS TABLE (media_id INTEGER, title TEXT)
+AS $$
+BEGIN
+  -- SEARCH HISTORY
+  INSERT INTO search_history (p_user_id, type, query)
+  VALUES (user_id, 'structured_string_search', 
```

Comment:
Should the user_id not be p_user_id - to have it consistent, or is this bound to the attribute user_id ?

INSERT INTO search_history (p_user_id, type, query)
 VALUES (user_id, 'structured_string_search',